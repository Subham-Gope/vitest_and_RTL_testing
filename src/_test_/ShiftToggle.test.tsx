// import { describe, test, expect } from "vitest";
// import { render, screen } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";
// import ShiftToggle from "../components/ShiftToggle";

// describe("Testing the ShiftToggle.tsx file", () => {
//   test("displays the entered text including {}", async () => {
//     // Setting up the userEvent
//     const user = userEvent.setup();

//     const { container } = render(
//       <ShiftToggle message={"Hello there"} num={5} done={true} />
//     );
//     expect(container).toMatchSnapshot();

//     const div = screen.getByText(/toggle/i);
//     expect(div).toBeInTheDocument();

//     // Initiating user interaction
//     div.focus();
//     await user.keyboard("{Shift>}{t}{/Shift}");
//     expect(div).toHaveTextContent("Toggle On");

//     await user.keyboard("[ShiftLeft>][KeyT][/ShiftLeft]");
//     expect(div).toHaveTextContent("Toggle Off");
//   });
// });
