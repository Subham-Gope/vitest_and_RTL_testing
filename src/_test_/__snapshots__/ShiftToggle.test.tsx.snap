// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the ShiftToggle.tsx file > displays the entered text including {} 1`] = `
<div>
  <div
    class="h-screen w-screen flex flex-col justify-center"
  >
    <h1>
      Message: 
      Hello there
      , Assignment Number: 
      5
      , Completion Status:
       
      true
    </h1>
    <div
      class="bg-orange-400 h-[400px] w-[400px] flex items-center justify-center text-2xl text-white font-bold m-auto"
      tabindex="0"
    >
      Toggle Off
    </div>
  </div>
</div>
`;
